### PR TITLE
Document usage of issues mailing list and remove obsolete archive links

### PR DESCRIPTION
**Pull request by zhangbutao**
1. Added issues mailing list
2. Remove the outdated incubator Archives link: `http://markmail.org/search/+list:org.apache.incubator.hive-xxxx`

### CODE REVIEW

File: content/community/mailingLists.md [hunk 1]
```diff
@@ -50,9 +48,16 @@ decisions, and vote on technical issues.
...
 * Post: <dev@hive.apache.org>
 * Unsubscribe: <dev-unsubscribe@hive.apache.org>
-* Archives:
-    * [Apache][dev_apache]
-    * [Markmail][dev_markmail]
+* Archives: [Apache][dev_apache]
+
+## Issues Mailing List
+---
+The issues list receives notifications about bugs and new features on [JIRA].
```
Comment (zabetak) [inline content/community/mailingLists.md:55]: "The issues list receives all notifications from the [JIRA] issue tracker."

**Review by zabetak — APPROVED** [review]
LGTM! 

Left one minor suggestion. 

Also the commit message could be improved to be more self contained. Eg., "Document usage of issues mailing list and remove obsolete archive links"

File: content/community/mailingLists.md (hunk 1 above)
Comment (zhangbutao) [reply content/community/mailingLists.md]: Done.Thx!

**Review by okumin — APPROVED** [review]
Thanks. I subscribed it

**Pull request merged**